template.c

```diff
@@ -0,0 +1,75 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+/* Expand template string using environment variables.
+ *
+ * If VAR1 is set to "abc" and VAR2 is not set, template
+ * "var1=${VAR1} var2=${VAR2} end" expands to
+ * "var1=abc var2= end".
+ */
+static void expand_template(char *tmpl, char *result)
+{
+	for (;;) {
+		char *start = strstr (tmpl, "${");
+		if (! start)
+			break;
+
+		char *end = strchr (start, '}');
+		if (! end)
+			break;
+
+		*start = '\0';
```

Comment:
Null terminal at start? Is it necessary?

Reply:
`start` points to the start of the template variable. Anything preceeding it should be copied verbatim to the output. And to avoid doing pointer acrobatics, I simply placed `NULL` at the end of the verbatim text and used `strcpy` to copy it later.